packages/ui-menu/src/Menu/index.tsx

```diff
@@ -231,87 +199,83 @@ class Menu extends Component<MenuProps> {
       this._activeSubMenu.hide(event)
       this._activeSubMenu = null
     }
   }
 
-  // @ts-expect-error ts-migrate(7006) FIXME: Parameter 'shown' implicitly has an 'any' type.
-  handleSubMenuToggle = (shown, subMenu) => {
+  handleSubMenuToggle: MenuProps['onToggle'] = (shown, subMenu) => {
     if (shown) {
       this._activeSubMenu = subMenu
     }
   }
 
-  // @ts-expect-error ts-migrate(7006) FIXME: Parameter 'event' implicitly has an 'any' type.
-  handleSubMenuDismiss = (event, documentClick) => {
-    if ((event && event.keyCode === keycode.codes.tab) || documentClick) {
+  handleSubMenuDismiss = (
+    event: React.MouseEvent | React.KeyboardEvent,
+    documentClick: boolean
+  ) => {
+    if (
+      (event && (event as React.KeyboardEvent).keyCode === keycode.codes.tab) ||
+      documentClick
+    ) {
       this.hide(event)
     }
   }
 
-  // @ts-expect-error ts-migrate(7006) FIXME: Parameter 'event' implicitly has an 'any' type.
-  hide = (event) => {
+  hide = (event: React.MouseEvent | React.KeyboardEvent) => {
     if (this._popover) {
-      // @ts-expect-error ts-migrate(2531) FIXME: Object is possibly 'null'.
-      this._popover.hide(event)
+      this._popover.hide(event as unknown as Event)
     }
   }
 
-  // @ts-expect-error ts-migrate(7006) FIXME: Parameter 'event' implicitly has an 'any' type.
-  show = (event) => {
+  show = (event?: React.MouseEvent | React.KeyboardEvent) => {
     if (this._popover) {
-      // @ts-expect-error ts-migrate(2531) FIXME: Object is possibly 'null'.
-      this._popover.show(event)
+      //TODO make events consistents
+      this._popover.show(event as unknown as Event)
     }
   }
 
   focus() {
     if (this.shown) {
-      // @ts-expect-error ts-migrate(2555) FIXME: Expected at least 5 arguments, but got 2.
-      error(this._menu && this._menu.focus, '[Menu] Could not focus the menu.')
-      // @ts-expect-error ts-migrate(2531) FIXME: Object is possibly 'null'.
-      this._menu.focus()
+      const hasFocusOnMenu =
+        this._menu !== null && typeof this._menu.focus === 'function'
+
+      error(hasFocusOnMenu, '[Menu] Could not focus the menu.')
+
+      if (hasFocusOnMenu) {
+        this._menu!.focus()
+      }
+      if (this._trigger !== null && typeof this._trigger.focus === 'function') {
+        this._trigger.focus()
+      }
     } else {
-      error(
-        // @ts-expect-error ts-migrate(2531) FIXME: Object is possibly 'null'.
-        this._trigger && this._trigger.focus,
-        '[Menu] Could not focus the trigger.'
-      )
-      // @ts-expect-error ts-migrate(2531) FIXME: Object is possibly 'null'.
-      this._trigger.focus()
+      error(false, '[Menu] Could not focus the trigger.')
     }
   }
 
   focused() {
     if (this.shown) {
       return containsActiveElement(this._menu) || this.state.hasFocus
     } else {
-      return containsActiveElement(this._trigger)
+      return false
```

Comment:
The original check is missing.

```suggestion
      return containsActiveElement(this._trigger)
```